transcom/mymove
# [MB-16228] Add migration to populate mto_shipments table for shipments without a destination address

## [Jira ticket](https://dp3.atlassian.net/browse/MB-16228)

## Summary

This PR adds a migration for HHG shipments that do not have a destination address to use the city/state/zip of their destination duty locations. This work does the backfill where [MB-16227](https://dp3.atlassian.net/browse/MB-16227) does the same for moves going forward. We should not merge this PR until we've completed and merged MB-16227.

We originally included all shipment types, but after feedback we reduced it to only HHGs per [this conversation](https://ustcdp3.slack.com/archives/C059Q014AM8/p1686586297980059).

### Verification Steps for the Author

These are to be checked by the author.


### How to test
Using your database management tool, copy and paste this query to get the list of entries that do not have a destination address:

```
SELECT mto_shipments.id, city, state, postal_code, country FROM mto_shipments
LEFT JOIN moves ON mto_shipments.move_id = moves.id
LEFT JOIN orders ON moves.orders_id = orders.id
LEFT JOIN duty_locations ON orders.new_duty_location_id = duty_locations.id
LEFT JOIN addresses ON duty_locations.address_id = addresses.id
WHERE destination_address_id IS NULL AND shipment_type = 'HHG'
``` 

Then run `make db_dev_migrate` and rerun the above query which should return no entries



[MB-16227]: https://dp3.atlassian.net/browse/MB-16227?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
